rpc/server.go

```diff
@@ -94,9 +95,12 @@ func NewServer(addr net.Addr, context *Context) *Server {
 // argument of the same type as `reqPrototype`. Both the argument and
 // return value of 'handler' should be a pointer to a protocol message
 // type. The handler function will be executed in a new goroutine.
-func (s *Server) Register(name string, handler func(proto.Message) (proto.Message, error),
+// If 'restricted' is true, only system users (RootUser and NodeUser)
+// may call this method.
+func (s *Server) Register(name string, restricted bool,
```

Comment:
Rather than a `restricted` argument here, how about leaving `Register` as-is and adding a `RegisterPublic` method for use by `sql`.